scipy/spatial/src/distance_wrap.c

```diff
@@ -32,6 +32,16 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+/* avoid unknown pragma warning in clang which may be used by developers */
+#ifndef __clang__
```

Comment:
one could add a large list of compilers that don't know gcc pragmas here, my thinking was silence it for the ones where we might actually get complaints, which are compilers developers are mostly using.
On others we have 3 more warnings in addition the already dozens from other places.
enabling it only for gcc macros is not so great because other compilers pretend to be gcc without being fully compatible (e.g. clang)

Also I'm hoping no compiler actually errors out on an unknown pragma but I think that is reasonable assumption.
Though gcc itself errors out on unknown gcc optimize option, so maybe a version 4 check is wise